### PR TITLE
Reduce the test noise by making id more realistic

### LINKED ISSUE #3893: ParseKey failures on CI
I've now seen two incidence of `ParseKey` (in mainProj) failing. Once in #3865 8 days ago, now in  #3875.

---

In #3865 it failed with:

```
[info] ! Key parser test.An explicitly specified axis is always parsed to that explicit value: Falsified after 51 passed tests.
[info] > Labels of failing property: 
[info] Key string: 'q / g'
[info] Structure: Env:
[info]     Tasks:
[info]     f (delegates: qq)
[info]     q (delegates: qq)
[info]     g (delegates: qq)
[info]     ka (delegates: qq)
[info]     someNfu (delegates: qq)
[info]     qq (delegates: )
[info] Build mfh:nam#cwhvfd :
[info]     Project q
[info]       Delegates:
[info]         ProjectRef(mfh:nam#cwhvfd,jqqdsp)
[info]       Configurations:
[info]         a
[info]         wz
[info]     Project jqqdsp
[info]       Delegates:
[info]         
[info]       Configurations:
[info]         a
[info]         wz
[info] current: ProjectRef(mfh:nam#cwhvfd,q)
[info] Settings:
[info] 	ProjectRef(uri("mfh:nam#cwhvfd"), "jqqdsp") / A / <key>
[info] 	   qq,ka,f,q,g,someNfu
[info] 
[info] 	ProjectRef(uri("mfh:nam#cwhvfd"), "jqqdsp") / A / qq / <key>
[info] 	   qq,ka,f,q,g,someNfu
[info] 
[info] 	ProjectRef(uri("mfh:nam#cwhvfd"), "jqqdsp") / A / someNfu / <key>
[info] 	   qq,ka,f,q,g,someNfu
[info] 
[info] 	ProjectRef(uri("mfh:nam#cwhvfd"), "jqqdsp") / Wz / <key>
[info] 	   qq,ka,f,q,g,someNfu
[info] 
[info] 	ProjectRef(uri("mfh:nam#cwhvfd"), "jqqdsp") / Wz / f / <key>
[info] 	   qq,ka,f,q,g,someNfu
[info] 
[info] 	ProjectRef(uri("mfh:nam#cwhvfd"), "jqqdsp") / Wz / g / <key>
[info] 	   qq,ka,f,q,g,someNfu
[info] 
[info] 	ProjectRef(uri("mfh:nam#cwhvfd"), "jqqdsp") / Wz / ka / <key>
[info] 	   qq,ka,f,q,g,someNfu
[info] 
[info] 	ProjectRef(uri("mfh:nam#cwhvfd"), "jqqdsp") / Wz / qq / <key>
[info] 	   qq,ka,f,q,g,someNfu
[info] 
[info] 	ProjectRef(uri("mfh:nam#cwhvfd"), "jqqdsp") / Wz / someNfu / <key>
[info] 	   qq,ka,f,q,g,someNfu
[info] 
[info] 	ProjectRef(uri("mfh:nam#cwhvfd"), "jqqdsp") / f / <key>
[info] 	   qq,ka,f,q,g,someNfu
[info] 
[info] 	ProjectRef(uri("mfh:nam#cwhvfd"), "jqqdsp") / qq / <key>
[info] 	   qq,ka,f,q,g,someNfu
[info] 
[info] 	ProjectRef(uri("mfh:nam#cwhvfd"), "q") / ka / <key>
[info] 	   qq,ka,f,q,g,someNfu
[info] 
[info] 	ProjectRef(uri("mfh:nam#cwhvfd"), "q") / someNfu / <key>
[info] 	   qq,ka,f,q,g,someNfu
[info] All keys:
[info] 	qq, ka, f, q, g, someNfu
[info] Expected: ProjectRef(uri("mfh:nam#cwhvfd"), "q") / q / g
[info] Scope(Select(ProjectRef(mfh:nam#cwhvfd,q)), Zero, Zero, Zero) == Scope(Select(ProjectRef(mfh:nam#cwhvfd,q)), Zero, Select(q), Zero): false
[info] ScopedKey(Scope(Select(ProjectRef(mfh:nam#cwhvfd,q)), Zero, Zero, Zero),g).key == ScopedKey(Scope(Select(ProjectRef(mfh:nam#cwhvfd,q)), Zero, Select(q), Zero),g).key: true
[info] Parsed: Right(ProjectRef(uri("mfh:nam#cwhvfd"), "q") / g)
[info] Mask: ScopeMask(false,false,true,false)
[info] Key: {mfh:nam#cwhvfd} / Wz / q / g
[info] > ARG_0: sbt.ParseKey$StructureKeyMask@626bae4c
```

---

In #3875 it failed with

```
! Key parser test.An unspecified project axis resolves to the current project: Falsified after 75 passed tests.
[info] > Labels of failing property: 
[info] Key string: 'a / m'
[info] Current: ProjectRef(pztl:v,lwSq)
[info] Parsed: Right(ProjectRef(uri("pztl:v"), "a") / m)
[info] Key: ProjectRef(uri("rwMsmGul:nrh#xu"), "nfCusdu") / Ol / a / m
[info] Mask: ScopeMask(false,false,true,false)
[info] Structure: Env:
[info]     Tasks:
[info]     a (delegates: zizv, db)
[info]     zizv (delegates: m, db)
[info]     m (delegates: db)
[info]     db (delegates: )
[info] Build pztl:v :
[info]     Project spgubHtn
[info]       Delegates:
[info]         ProjectRef(pztl:v,qB)
[info]         ProjectRef(pztl:v,lwSq)
[info]       Configurations:
[info]         ol
[info]     Project j
[info]       Delegates:
[info]         ProjectRef(pztl:v,lwSq)
[info]         ProjectRef(pztl:v,qB)
[info]       Configurations:
[info]         ol
[info]     Project gdjE
[info]       Delegates:
[info]         ProjectRef(pztl:v,a)
[info]         ProjectRef(pztl:v,lwSq)
[info]       Configurations:
[info]         ol
[info]     Project a
[info]       Delegates:
[info]         ProjectRef(pztl:v,btQpaK)
[info]         ProjectRef(pztl:v,lwSq)
[info]       Configurations:
[info]         ol
[info]     Project btQpaK
[info]       Delegates:
[info]         ProjectRef(pztl:v,qB)
[info]         ProjectRef(pztl:v,lwSq)
[info]       Configurations:
[info]         ol
[info]     Project qB
[info]       Delegates:
[info]         ProjectRef(pztl:v,lwSq)
[info]       Configurations:
[info]         ol
[info]     Project lwSq
[info]       Delegates:
[info]         
[info]       Configurations:
[info]         ol
[info]   Build rwMsmGul:nrh#xu :
[info]     Project nfCusdu
[info]       Delegates:
[info]         
[info]       Configurations:
[info]         ol
[info] current: ProjectRef(pztl:v,lwSq)
[info] Settings:
[info] 	Global / <key>
[info] 	   m,a,zizv,db
[info] 
[info] 	ProjectRef(uri("pztl:v"), "a") / <key>
[info] 	   a,m,zizv,db
[info] 
[info] 	ProjectRef(uri("pztl:v"), "a") / Ol / <key>
[info] 	   db,a,m,zizv
[info] 
[info] 	ProjectRef(uri("pztl:v"), "a") / Ol / db / <key>
[info] 	   m,db,zizv,a
[info] 
[info] 	ProjectRef(uri("pztl:v"), "a") / Ol / zizv / <key>
[info] 	   zizv,a,db,m
[info] 
[info] 	ProjectRef(uri("pztl:v"), "a") / a / <key>
[info] 	   zizv,db,a,m
[info] 
[info] 	ProjectRef(uri("pztl:v"), "a") / db / <key>
[info] 	   m,a,db,zizv
[info] 
[info] 	ProjectRef(uri("pztl:v"), "btQpaK") / Ol / <key>
[info] 	   zizv,db,m,a
[info] 
[info] 	ProjectRef(uri("pztl:v"), "btQpaK") / Ol / a / <key>
[info] 	   db,a,zizv,m
[info] 
[info] 	ProjectRef(uri("pztl:v"), "btQpaK") / Ol / db / <key>
[info] 	   zizv,a,db,m
[info] 
[info] 	ProjectRef(uri("pztl:v"), "btQpaK") / Ol / m / <key>
[info] 	   zizv,a,db,m
[info] 
[info] 	ProjectRef(uri("pztl:v"), "btQpaK") / Ol / zizv / <key>
[info] 	   a,zizv,m,db
[info] 
[info] 	ProjectRef(uri("pztl:v"), "btQpaK") / db / <key>
[info] 	   m,a,db,zizv
[info] 
[info] 	ProjectRef(uri("pztl:v"), "btQpaK") / m / <key>
[info] 	   zizv,a,db,m
[info] 
[info] 	ProjectRef(uri("pztl:v"), "btQpaK") / zizv / <key>
[info] 	   a,m,zizv,db
[info] 
[info] 	ProjectRef(uri("pztl:v"), "gdjE") / <key>
[info] 	   db,m,a,zizv
[info] 
[info] 	ProjectRef(uri("pztl:v"), "gdjE") / a / <key>
[info] 	   db,m,a,zizv
[info] 
[info] 	ProjectRef(uri("pztl:v"), "gdjE") / m / <key>
[info] 	   zizv,db,a,m
[info] 
[info] 	ProjectRef(uri("pztl:v"), "j") / Ol / db / <key>
[info] 	   db,a,zizv,m
[info] 
[info] 	ProjectRef(uri("pztl:v"), "j") / Ol / zizv / <key>
[info] 	   db,m,a,zizv
[info] 
[info] 	ProjectRef(uri("pztl:v"), "j") / a / <key>
[info] 	   db,m,a,zizv
[info] 
[info] 	ProjectRef(uri("pztl:v"), "j") / db / <key>
[info] 	   db,a,zizv,m
[info] 
[info] 	ProjectRef(uri("pztl:v"), "j") / m / <key>
[info] 	   m,zizv,db,a
[info] 
[info] 	ProjectRef(uri("pztl:v"), "lwSq") / Ol / <key>
[info] 	   zizv,m,a,db
[info] 
[info] 	ProjectRef(uri("pztl:v"), "lwSq") / Ol / db / <key>
[info] 	   db,zizv,m,a
[info] 
[info] 	ProjectRef(uri("pztl:v"), "lwSq") / Ol / m / <key>
[info] 	   m,zizv,a,db
[info] 
[info] 	ProjectRef(uri("pztl:v"), "lwSq") / Ol / zizv / <key>
[info] 	   a,m,db,zizv
[info] 
[info] 	ProjectRef(uri("pztl:v"), "lwSq") / a / <key>
[info] 	   db,zizv,m,a
[info] 
[info] 	ProjectRef(uri("pztl:v"), "lwSq") / db / <key>
[info] 	   zizv,db,m,a
[info] 
[info] 	ProjectRef(uri("pztl:v"), "lwSq") / m / <key>
[info] 	   db,a,m,zizv
[info] 
[info] 	ProjectRef(uri("pztl:v"), "lwSq") / zizv / <key>
[info] 	   a,zizv,db,m
[info] 
[info] 	ProjectRef(uri("pztl:v"), "qB") / <key>
[info] 	   m,zizv,db,a
[info] 
[info] 	ProjectRef(uri("pztl:v"), "qB") / Ol / a / <key>
[info] 	   m,zizv,db,a
[info] 
[info] 	ProjectRef(uri("pztl:v"), "qB") / Ol / db / <key>
[info] 	   db,a,m,zizv
[info] 
[info] 	ProjectRef(uri("pztl:v"), "qB") / Ol / zizv / <key>
[info] 	   a,m,db,zizv
[info] 
[info] 	ProjectRef(uri("pztl:v"), "qB") / db / <key>
[info] 	   db,a,zizv,m
[info] 
[info] 	ProjectRef(uri("pztl:v"), "qB") / zizv / <key>
[info] 	   db,a,m,zizv
[info] 
[info] 	ProjectRef(uri("pztl:v"), "spgubHtn") / Ol / <key>
[info] 	   a,zizv,m,db
[info] 
[info] 	ProjectRef(uri("pztl:v"), "spgubHtn") / Ol / a / <key>
[info] 	   db,m,a,zizv
[info] 
[info] 	ProjectRef(uri("pztl:v"), "spgubHtn") / Ol / db / <key>
[info] 	   db,m,zizv,a
[info] 
[info] 	ProjectRef(uri("pztl:v"), "spgubHtn") / m / <key>
[info] 	   db,zizv,a,m
[info] 
[info] 	ProjectRef(uri("rwMsmGul:nrh#xu"), "nfCusdu") / <key>
[info] 	   db,m,zizv,a
[info] 
[info] 	ProjectRef(uri("rwMsmGul:nrh#xu"), "nfCusdu") / Ol / <key>
[info] 	   zizv,m,a,db
[info] 
[info] 	ProjectRef(uri("rwMsmGul:nrh#xu"), "nfCusdu") / Ol / a / <key>
[info] 	   db,zizv,a,m
[info] 
[info] 	ProjectRef(uri("rwMsmGul:nrh#xu"), "nfCusdu") / Ol / db / <key>
[info] 	   m,a,zizv,db
[info] 
[info] 	ProjectRef(uri("rwMsmGul:nrh#xu"), "nfCusdu") / Ol / m / <key>
[info] 	   db,a,zizv,m
[info] 
[info] 	ProjectRef(uri("rwMsmGul:nrh#xu"), "nfCusdu") / Ol / zizv / <key>
[info] 	   db,m,a,zizv
[info] 
[info] 	ProjectRef(uri("rwMsmGul:nrh#xu"), "nfCusdu") / a / <key>
[info] 	   db,a,zizv,m
[info] 
[info] 	ProjectRef(uri("rwMsmGul:nrh#xu"), "nfCusdu") / db / <key>
[info] 	   db,a,zizv,m
[info] 
[info] 	ProjectRef(uri("rwMsmGul:nrh#xu"), "nfCusdu") / m / <key>
[info] 	   m,db,zizv,a
[info] 
[info] 	ProjectRef(uri("rwMsmGul:nrh#xu"), "nfCusdu") / zizv / <key>
[info] 	   a,zizv,m,db
[info] 
[info] 	Zero / Ol / db / <key>
[info] 	   zizv,a,m,db
[info] 
[info] 	Zero / Ol / m / <key>
[info] 	   db,a,zizv,m
[info] 
[info] 	Zero / a / <key>
[info] 	   a,db,zizv,m
[info] 
[info] 	Zero / db / <key>
[info] 	   db,m,zizv,a
[info] 
[info] 	Zero / m / <key>
[info] 	   a,db,zizv,m
[info] 
[info] 	Zero / zizv / <key>
[info] 	   a,db,m,zizv
[info] All keys:
[info] 	a, m, db, zizv
[info] > ARG_0: sbt.ParseKey$StructureKeyMask@2421d6b0
```

### PR DESCRIPTION
**Pull request by eed3si9n**
Fixes #3893

First increase the ScalaCheck run from 100 to 10000 to reduce the flakiness. If there's a failure, this will now fail all the time. 
Then I attempted to fix the flaky ParserSpec by making the id generation more realistic ASCII identifiers.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -596,34 +616,47 @@ lazy val zincLmIntegrationProj = (project in file("zinc-lm-integration"))
...
+        Seq(launcherInterface) ++
+        log4jDependencies ++
+        Seq(scalaCacheCaffeine, lmCoursierShaded)
     },
     Compile / scalacOptions -= "-Xfatal-warnings",
     managedSourceDirectories in Compile +=
       baseDirectory.value / "src" / "main" / "contraband-scala",
     sourceManaged in (Compile, generateContrabands) := baseDirectory.value / "src" / "main" / "contraband-scala",
+    testOptions in Test += Tests
+      .Argument(TestFrameworks.ScalaCheck, "-minSuccessfulTests", "10000"),
```
Comment (eed3si9n) [inline build.sbt]: The only real change in build.sbt file is this line.

**Comment by eed3si9n** [comment]
```
[info] + Key parser test.An unspecified task axis resolves to Zero: OK, passed 10000 tests.
[info] Elapsed time: 4 min 20.308 sec 
[info] + Key parser test.An explicitly specified axis is always parsed to that explicit value: OK, passed 10000 tests.
[info] Elapsed time: 4 min 39.504 sec 
[info] + Key parser test.An unspecified project axis resolves to the current project: OK, passed 10000 tests.
[info] Elapsed time: 4 min 6.673 sec 
[info] + Key parser test.An unspecified configuration axis resolves to the first configuration directly defining the key or else Zero: OK, passed 10000 tests.
[info] Elapsed time: 4 min 15.355 sec 
```

Adding 17 minutes is a bit too much. Maybe I should reduce the number of ScalaCheck run once it passes Travis CI.

**Comment by dwijnand** [comment]
Do you know what causes the test to fail?

Also, watch out not to break the invariants when the generators shrink.

**Comment by eed3si9n** [comment]
I am not sure when the identifiers started including non-ASCII, but my guess is that it's been flaky since then. It's just that the sample size was too small to hit it often, but it seems that Dbuild hits it more frequently than Travis CI (though I've seen it on Travis a number of times).

**Comment by dwijnand** [comment]
Maybe an upstream ScalaCheck. change.  Or maybe it always used non-ASCII but the test can't shrink down to ASCII.

**Comment by eed3si9n** [comment]
It's entirely possible that newer ScalaCheck is better at coming up with breaking examples, and the failures are legit. I did use a lot of `listOfN` as opposed to `listOf` which can produce single letter identifiers that are prone to get duplicates. Maybe I can circle back later, but for now I just want the build to catch new / 1.3.x bugs.

**Pull request merged**